WolvenKit.App/ViewModels/Documents/RedDocumentViewModel.cs

```diff
@@ -189,74 +190,15 @@ public override async Task Save(object? parameter)
         await Task.CompletedTask;
     }
 
-    public void SaveSync(object? parameter)
+    public void SaveSync(object? _)
     {
-        var ms = new MemoryStream();
-        var file = GetMainFile();
-
-        try
-        {
-            if (file is not null && Cr2wFile != null)
-            {
-                var cr2w = Cr2wFile;
-                if (_hookService is AppHookService appHookService && !appHookService.OnSave(FilePath, ref cr2w))
-                {
-                    _loggerService.Error($"Error while processing onSave hooks");
-                    return;
-                }
-
-                SaveHashedValues(cr2w);
-
-                using var writer = new CR2WWriter(ms, Encoding.UTF8, true) { LoggerService = _loggerService };
-                writer.WriteFile(cr2w);
-            }
-        }
-        catch (Exception e)
+        if (GetMainFile() is null || !DocumentTools.SaveCr2W(Cr2wFile, FilePath))
         {
-            _loggerService.Error($"Error while saving {FilePath}");
-            _loggerService.Error(e);
-
             return;
         }
 
-        if (FileHelper.SafeWrite(ms, FilePath, _loggerService))
-        {
-            LastWriteTime = File.GetLastWriteTime(FilePath);
-        }
-
+        LastWriteTime = File.GetLastWriteTime(FilePath);
         SetIsDirty(false);
-        _loggerService.Success($"Saved file {FilePath}");
-    }
-
-    private void SaveHashedValues(CR2WFile file)
-    {
-        if (_hashService is not HashServiceExt hashService)
-        {
-            return;
-        }
-
-        foreach (var (path, value) in file.RootChunk.GetEnumerator())
-        {
-            if (value is IRedRef redRef && redRef.DepotPath != ResourcePath.Empty)
-            {
-                if (!redRef.DepotPath.TryGetResolvedText(out var refPath))
-                {
-                    continue;
-                }
-
-                hashService.AddResourcePath(refPath);
-            }
-
-            if (value is TweakDBID tweakDbId && tweakDbId != TweakDBID.Empty)
-            {
-                if (!tweakDbId.TryGetResolvedText(out var tweakName))
-                {
-                    continue;
-                }
-
-                hashService.AddTweakName(tweakName);
-            }
-        }
     }
 
     public override void SaveAs(object parameter) => throw new NotImplementedException();
```

Comment:
Could handle your SaveAs here (as intended).
Currently not sure what parameter is used for, but could also add a filename arg and use the instead of `FilePath`.
Could even refactor `Save` to call `SaveAs(parameter, FilePath)` then.

Reply:
I figured I'd put it into the interface at the top level. Didn't think much about it. Can you elaborate on your other idea?

Reply:
1. Move the code from `Save(object parameter)` to `SaveAs(object parameter)`
2. Change `SaveAs(object parameter)` to `SaveAs(object parameter, string targetPath)`
3. Replace `FilePath` with `targetPath` in the code
4. Change `Save(object parameter)` to `Save(object parameter) => SaveAs(parameter, FilePath);`